packages/typescript/lib/resolveModuleName.ts

```diff
@@ -52,6 +52,24 @@ export function createResolveModuleName<T>(
 						}
 					}
 				}
+				if (typescript.resolveHiddenExtensions && fileName.endsWith(`.d.ts`)) {
```

Comment:
Just for understanding/posterity: I was confused about why we'd be checking `.d.ts` extension here, given that for an Ember .gts file, there wouldn't be a `.d.ts` file on disk. But the reason this works is that TS checks for xxx.js, xxx.ts, and then xxx.d.ts. We hook into the check for `.d.ts` and take that opportunity to check for extra file extensions (such as .gts)